woocommerce/woocommerce-admin
# Add plugin and jetpack wc-api methods to shipping task

Fixes #2482

* Adds plugin installation and activation methods to the wc-api.
* Adds jetpack connection URL methods to the wc-api.
* Adds the shipping label and connection steps to the shipping task in the task list.

I think in a follow-up, we'll want to grab Jetpack connection state and skip/hide this step if already connected.

### Screenshots

<img width="710" alt="Screen Shot 2019-08-12 at 2 42 40 PM" src="https://user-images.githubusercontent.com/10561050/62848771-91377280-bd0f-11e9-83b0-2a1043acf6ce.png">
<img width="709" alt="Screen Shot 2019-08-12 at 2 42 35 PM" src="https://user-images.githubusercontent.com/10561050/62848772-91377280-bd0f-11e9-86b9-a6241762fd5d.png">

### Detailed test instructions:

1.  Enable the task list by setting `const requiredTasksComplete = false;` in dashboard/index.js.
2.  Go to the shipping step and complete the shipping rates portion.
3.  Make sure that step 3 ("Enable shipping label printing") installs and activates Jetpack and WooCommerce services whether currently installed or not.
4.  Make sure step 4 ("Connect your store") redirects to the Jetpack connection screen on click.
5.  Make sure the plugins step of the profiler continues to work as expected `wp-admin/admin.php?page=wc-admin&step=plugins`.
6.  Make sure that button states are "Busy" when activation/installation is being requested and double activation/installation isn't possible.
